src/main/java/bio/terra/common/logging/GoogleJsonLayout.java

```diff
@@ -0,0 +1,263 @@
+package bio.terra.common.logging;
+
+import ch.qos.logback.classic.pattern.ThrowableProxyConverter;
+import ch.qos.logback.classic.spi.CallerData;
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import ch.qos.logback.contrib.jackson.JacksonJsonFormatter;
+import ch.qos.logback.contrib.json.JsonLayoutBase;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.cloud.ServiceOptions;
+import com.google.gson.JsonObject;
+import io.opencensus.trace.SpanId;
+import io.opencensus.trace.TraceId;
+import io.opencensus.trace.Tracer;
+import io.opencensus.trace.Tracing;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.util.StringUtils;
+
+/**
+ * A Logback JSON layout aimed at generating log lines with useful details for ingestion by the
+ * Google Cloud Logging system.
+ *
+ * <p>See the LoggingTest integration test for more details on the output structure, including
+ * concrete examples of field values.
+ *
+ * <p>The output JSON relies heavily on Google Cloud's interpretation of structured JSON logs,
+ * including so-called "special fields in JSON payloads". See
+ * https://cloud.google.com/logging/docs/structured-logging#special-payload-fields for more details.
```

Comment:
Make this a live link? (same request for other links in javadocs, but I won't repeat it)
```suggestion
 * including so-called "special fields in JSON payloads". See <a
 * href="https://cloud.google.com/logging/docs/structured-logging#special-payload-fields">Google's
 * documentation</a> for more details.
```

Reply:
I don't have super strong feelings about this — as long as it's easy enough to click live links from within IntelliJ, since I almost never read Javadoc except from there.

Reply:
I intend to use Stairway repo to build a generate GHA to post javadocs on github pages. That should make them more useful and worth debugging (and adding live links...)

Reply:
So in intellij, without wrapping in an explicit `<a>`, the link is command clickable, but only if the documentation is displayed raw. If it's displaying the nicely formatted version, the `<a>` version is clickable, and the regular version isn't. FWIW I also only view it in intellij, mine just defaults to the nicely formatted version of javadoc comments, there should be a little toggle button in the margin.